code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -78,6 +81,72 @@
 	if(world.time <= next_move)
 		return
 	A.attack_ghost(src)
+	if(ishuman(A) && in_range(src, A))
+		src.Harvest(A)
```

Comment:
if ```src``` is the revenant (which is most of these cases I believe it is) then ```src``` is implied, please remove all ```src``` where possible.